### PR TITLE
Enhance Dockerfile adding multistage build

### PR DESCRIPTION
**Pull request by sergioasantiago**

### CODE REVIEW

**Comment by ribbybibby** [comment]
Thanks for the PR!

The Dockerfile is designed to work with goreleaser which we use to produce our releases. This is why it copies the binary into the container.

That being said, I do think it is valuable to have a Dockerfile that you can build with `docker build` when testing. So perhaps, in addition to the changes in this PR, we should rename the current Dockerfile to Dockerfile.goreleaser and modify goreleaser to use that.

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+
+RUN go mod download
+
+COPY *.go ./
+
+COPY internal ./internal
+
+RUN go build -o /dependency-track-exporter
+
+FROM gcr.io/distroless/base-debian11:latest-amd64
```
Comment (ribbybibby) [inline Dockerfile:17]: Could we use `gcr.io/distroless/static:nonroot` here? Unless there's a reason we need debian, I'd prefer to use the most minimal possible image.

**Review by ribbybibby — CHANGES_REQUESTED** [review]
One comment about the base image. And, as mentioned, we should use the existing Dockerfile with `goreleaser`.

**Comment by sergioasantiago** [comment]
> One comment about the base image. And, as mentioned, we should use the existing Dockerfile with `goreleaser`.

Thanks for your comment, Changes have been made.

**Review by ribbybibby — APPROVED** [review]

**Pull request merged**